Yosemite/YosemiteTests/Stores/ShippingLabelStoreTests.swift

```diff
@@ -38,6 +38,130 @@ final class ShippingLabelStoreTests: XCTestCase {
+        let expectedResponse = OrderShippingLabelListResponse(shippingLabels: [expectedShippingLabel], settings: expectedSettings)
+        remote.whenLoadingShippingLabels(siteID: sampleSiteID, orderID: orderID, thenReturn: .success(expectedResponse))
+        let store = ShippingLabelStore(dispatcher: dispatcher, storageManager: storageManager, network: network, remote: remote)
+
+        insertOrder(siteID: sampleSiteID, orderID: orderID)
+
+        // When
+        let result: Result<Void, Error> = try waitFor { promise in
+            let action = ShippingLabelAction.synchronizeShippingLabels(siteID: self.sampleSiteID, orderID: orderID) { result in
+                promise(result)
+            }
+            store.onAction(action)
+        }
+
+        // Then
+        XCTAssertNoThrow(try XCTUnwrap(result.get()))
+
+        let persistedOrder = try XCTUnwrap(viewStorage.loadOrder(siteID: sampleSiteID, orderID: orderID))
+        let persistedShippingLabels = try XCTUnwrap(viewStorage.loadAllShippingLabels(siteID: sampleSiteID, orderID: orderID))
+        XCTAssertEqual(persistedOrder.shippingLabels, Set(persistedShippingLabels))
+        XCTAssertEqual(persistedShippingLabels.map { $0.toReadOnly() }, [expectedShippingLabel])
+
+        let persistedSettings = try XCTUnwrap(viewStorage.loadShippingLabelSettings(siteID: sampleSiteID, orderID: orderID))
+        XCTAssertEqual(persistedOrder.shippingLabelSettings, persistedSettings)
+        XCTAssertEqual(persistedSettings.toReadOnly(), expectedSettings)
+    }
+
+    func test_loadShippingLabels_does_not_persist_shipping_labels_and_settings_on_success_with_empty_shipping_labels() throws {
+        // Given
+        let remote = MockShippingLabelRemote()
+        let orderID: Int64 = 22
+        let expectedSettings = Yosemite.ShippingLabelSettings(siteID: sampleSiteID, orderID: orderID, paperSize: .letter)
+        // The response has no shipping labels but settings, to simulate an order without shipping labels.
+        let expectedResponse = OrderShippingLabelListResponse(shippingLabels: [], settings: expectedSettings)
+        remote.whenLoadingShippingLabels(siteID: sampleSiteID, orderID: orderID, thenReturn: .success(expectedResponse))
+        let store = ShippingLabelStore(dispatcher: dispatcher, storageManager: storageManager, network: network, remote: remote)
+
+        insertOrder(siteID: sampleSiteID, orderID: orderID)
+
+        // When
+        let result: Result<Void, Error> = try waitFor { promise in
+            let action = ShippingLabelAction.synchronizeShippingLabels(siteID: self.sampleSiteID, orderID: orderID) { result in
+                promise(result)
+            }
+            store.onAction(action)
+        }
+
+        // Then
+        XCTAssertNoThrow(try XCTUnwrap(result.get()))
+
+        let persistedOrder = try XCTUnwrap(viewStorage.loadOrder(siteID: sampleSiteID, orderID: orderID))
+        XCTAssertEqual(persistedOrder.shippingLabels ?? [], [])
```

Comment:
I wonder if we should have made `persistedOrder.shippingLabels` non-nil and default to an empty `Set`. 🤔

Reply:
I actually set `Order.shippingLabels` relationship to be non-optional first, but then encountered a crash since the relationship is uninitialized/unknown. I didn't find a way to set a default value in Xcode (screenshot below), we might have to do a manual migration to set the default value if we really want a non-optional to-many relationship 😐 

<img width="343" alt="Screen Shot 2020-11-20 at 11 12 12 AM" src="https://user-images.githubusercontent.com/1945542/99753634-8d54eb80-2b21-11eb-954d-3d8ab10672b4.png">

Reply:
Oh, that's right. It's fine then. 🙂